webrtc/sdp-offer-dc.html

```diff
@@ -0,0 +1,110 @@
+<!doctype html>
+<!--
+This test uses data only, and thus does not require fake media devices.
+-->
+
+<html>
+<head>
+  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+  <title>SDP generated by createOffer</title>
+</head>
+<body>
+  <div id="log"></div>
+
+  <!-- These files are in place when executing on W3C. -->
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/common/vendor-prefix.js"
+           data-prefixed-objects=
+        '[{"ancestors":["window"], "name":"RTCPeerConnection"}]'></script>
+  <script type="text/javascript">
+  var test = async_test('The SDP generated by createOffer matches JSEP requirements for initial offer.');
+
+  var gFirstConnection = null;
+
+  // This function starts the test.
+  test.step(function() {
+    gFirstConnection = new RTCPeerConnection(null);
+
+    gFirstConnection.createDataChannel('channel');
+
+    gFirstConnection.createOffer()
+      .then(test.step_func(function(offer) {
+      assert_equals(offer.type, "offer");
+      var sdp = offer.sdp;
+      var sdp_lines = sdp.split("\r\n");
+      assert_equals(sdp_lines[0], "v=0", "The first SDP line MUST be 'v=0'");
+      var origin = sdp_lines[1];
+      var origin_comp = origin.split(" ");
+      assert_equals(origin_comp[0], "-", 'The value of the <username> field SHOULD be "-"');
+      assert_equals(origin_comp[2], "0", 'The value of the <sess-version> field SHOULD be zero');
+      assert_equals(sdp_lines[2], "s=-", 'The third SDP line MUST be a "s=" line.  a single dash SHOULD be used as the session name');
+      var hasTLine = false;
+      var m_sections = sdp.split(/m=/);
+      var session_lines = m_sections[0].split("\r\n");
+      for (var i = 0 ; i < session_lines.length ; i++) {
+         // Start of a m= line, stop
+         if (session_lines[i].match(/^m=/)) {
+            break;
+         }
+         if (session_lines[i].match(/^[iuepbrz]=/)) {
+            assert_unreached('Session Information ("i="), URI ("u="), Email Address ("e="), Phone Number ("p="), Bandwidth ("b="), Repeat Times ("r="), and Time Zones ("z=") lines are not useful in this context and SHOULD NOT be included.');
+         }
+         if (session_lines[i].match(/^k=/)) {
+             assert_unreached('Encryption Keys ("k=") lines do not provide sufficient security and MUST NOT be included.');
+         }
+         if (session_lines[i] === 't=0 0') {
+            hasTLine = true;
+         }
+      }
+      assert_true(hasTLine, 'A "t=" line MUST be added; both <start-time> and <stop-time> SHOULD be set to zero');
+      assert_equals(m_sections.length, 2, "One m= section per stream");
```

Comment:
This comment is confusing, since you will only have one m= section. Suggest doing "session_level = m_sections.pop()" above, and adding a comment "What comes before the first m= line is session level data".